plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -626,7 +666,18 @@ enterComputeCek costs = computeCek where
       case resultOrErr of
           -- Turn the cause of a possible failure, being a 'CekValue', into a 'Term'.
           Left e       -> throwCek $ mapCauseInMachineException (void . dischargeCekValue) e
-          Right result -> returnCek ctx result
+          Right result -> returnCek unbudgetedSteps ctx result
+
+    -- | Spend the budget that has been accumulated for a number of machine steps.
+    spendAccumulatedBudget :: Int -> CekM s ()
+    spendAccumulatedBudget !unbudgetedSteps = spendBudgetCek BStep (stimes unbudgetedSteps (cekStepCost ?cekCosts))
+
+    -- | Spend the budget that has accumulated for a number of machine steps, but only if we've exceeded our slippage.
+    maybeSpendBudget :: Int -> CekM s Int
+    maybeSpendBudget !unbudgetedSteps =
+        if unbudgetedSteps >= ?cekSlippage
+        then spendAccumulatedBudget unbudgetedSteps >> pure 0
+        else pure unbudgetedSteps
```

Comment:
Have you tried putting an `INLINE` pragma on that one?

And what happens if you return `pure 1` / `pure $! unbudgetedSteps + 1` and drop all the `+ 1` down the line as well as all the bangs?

Reply:
I didn't think inlining this one would actually be good... maybe it would be!

Reply:
Okay, doing this lets me remove the `+1`s, which is good. It lets me remove the bangs on the patterns for the results of calling this function, but not the bangs on the arguments to the main recursive functions. If I remove those we get lots of reboxing again.

Reply:
And, finally, after recent changes, this function inlines and `spendAccumulatedBudget` does not, which finally removes literally all of the boxing related to this parameter!